From 24bb7675f75260b677789f2a406aff5883f6b370 Mon Sep 17 00:00:00 2001
From: Elliot Chen <elliot.chen@nxp.com>
Date: Thu, 6 Apr 2023 10:40:25 +0800
Subject: [PATCH 63/91] MMFMWK-9219 decodebin3: can not remove input stream if
 it is eos when handling pending input

For some streams whose duration is small, should not remove
input stream if it is eos when handling pending input, otherwise
it can not guarantee that decodebin3 can send out eos event successfully

Upstream-Status: Pending

Signed-off-by: Elliot Chen <elliot.chen@nxp.com>
Signed-off-by: Xiaolei Wang <xiaolei.wang@windriver.com>
---
 gst/playback/gstdecodebin3.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/gst/playback/gstdecodebin3.c b/gst/playback/gstdecodebin3.c
index 666130f..2e37851 100644
--- a/gst/playback/gstdecodebin3.c
+++ b/gst/playback/gstdecodebin3.c
@@ -1333,12 +1333,10 @@ gst_decodebin_input_unblock_streams (DecodebinInput * input,
       input_stream->buffer_probe_id = 0;
     }
 
-    if (input_stream->saw_eos) {
-      GST_DEBUG_OBJECT (dbin, "Removing EOS'd stream");
-      remove_input_stream (dbin, input_stream);
-      tmp = dbin->input_streams;
-    } else
-      tmp = next;
+    /* For some streams whose duration is small, should not remove
+     * input stream if it's EOS, otherwise it can't guarantee that decodebin3
+     * can send out eos event successfully in multiqueue_src_probe function */
+    tmp = next;
   }
 
   /* Weed out unused multiqueue slots */
-- 
2.43.0

